server/auth/auth.go

```diff
@@ -826,6 +559,51 @@ func (auth *AuthManager) handleConnect(conn comms.Link, msg *msgjson.Message) *m
 		}
 	}
 
+	// Check to see if there is already an existing client for this account.
+	respHandlers := make(map[uint64]*respHandler)
+	client := auth.user(acctInfo.ID)
+	if client != nil {
+		// Disconnect and remove known connections. We are creating a new one,
+		// but persist the response handlers.
```

Comment:
We might not create a new connection for this user if some error occurs downline before the call to `auth.addClient(client)`. Is this intentional? 

Oh I see that this was moved from below, I understand the reason for moving the account disabling check up here but is it necessary to remove any existing connection earlier as well?

Reply:
I've refactored this a bit. See 0d6c83d